Firestore/core/src/model/mutable_document.cc

```diff
@@ -63,7 +59,7 @@ MutableDocument& MutableDocument::ConvertToNoDocument(
     const SnapshotVersion& version) {
   version_ = version;
   document_type_ = DocumentType::kNoDocument;
-  value_ = {};
+  value_ = std::make_shared<ObjectValue>();
```

Comment:
You want to make sure `value_` is never null to avoid checking it before each access, right?